apps/aehttp/test/aehttp_integration_SUITE.erl

```diff
@@ -906,10 +922,19 @@ block_txs_count_pending(_Config) ->
     rpc(aec_conductor, start_mining, []),
     timer:sleep(100),% so the miner is started
     add_spend_txs(),
-    {ok, PendingBlock} = get_pending_block(),
-    TxsCount = length(aec_blocks:txs(PendingBlock)), 
-    {ok, 200, #{<<"count">> := TxsCount}} =
-        get_block_txs_count_preset("pending"),
+    GetPending =
+        fun TryToGet(Attempts) when Attempts < 1 ->
+                {error, no_pending_block_returned};
+            TryToGet(Attempts) ->
+                case get_block_txs_count_preset("pending") of
+                    {ok, 200, P} -> {ok, P};
+                    {ok, 404, _} -> TryToGet(Attempts -1)
+                end
+        end,
+    {ok, Txs} = rpc(aec_tx_pool, peek, [infinity]),
+    ct:log("Txs count in pool: ~p", [length(Txs)]),
+    {ok, #{<<"count">> := TxsCount}} = GetPending(10),
+    true = TxsCount =< length(Txs),
```

Comment:
It appears to me that by the time "aec_tx_pool:peek" is called, all txs may have been already removed from mempool. Hence this line may be executed as`true = 1 =< 0` that causes badmatch exception.

May I suggest you describe that this test aims at testing so to simplify the logic?

Reply:
Removing txs from mempool happens when a new block is mined. This test expects that no block is mined durring test's execution. For this we mine the first X blocks (that have no difficulty readjustment) and then set the expected mine rate to be one block per hour. Taken into account the previous X blocks and the high mine rate should make the next block really harder to mine.